api/apps/api/src/modules/scenarios/planning-units/scenario-planning-units-linker-service.ts

```diff
@@ -0,0 +1,65 @@
+import { Injectable, Logger } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Repository } from 'typeorm';
+
+import { DbConnections } from '@marxan-api/ormconfig.connections';
+import { ScenariosPlanningUnitGeoEntity } from '@marxan/scenarios-planning-unit';
+import {
+  PlanningUnitGridShape,
+  Project,
+} from '@marxan-api/modules/projects/project.api.entity';
+import { Scenario } from '../scenario.api.entity';
+import { isNil } from 'lodash';
+
+@Injectable()
+export class ScenarioPlanningUnitsLinkerService {
+  constructor(
+    @InjectRepository(
+      ScenariosPlanningUnitGeoEntity,
+      DbConnections.geoprocessingDB,
+    )
+    private readonly puRepo: Repository<ScenariosPlanningUnitGeoEntity>,
+    @InjectRepository(Project)
+    private readonly projectsRepo: Repository<Project>,
+  ) {}
+
+  /** Currently we only support linking regular planning unit geometries
+   * computed via ST_HexagonGrid or ST_SquareGrid */
+  private isPlanningUnitGridShapeSupported(
+    shape: PlanningUnitGridShape,
+  ): boolean {
+    return [
+      PlanningUnitGridShape.hexagon,
+      PlanningUnitGridShape.square,
+    ].includes(shape);
+  }
+
+  /**
+   * Link (existing) planning unit geometries matching project settings and
+   * falling within the project bbox to the scenario being created.
+   */
+  async link(scenario: Scenario): Promise<void> {
+    const project = await this.projectsRepo.findOneOrFail(scenario.projectId);
+    if (
+      !this.isPlanningUnitGridShapeSupported 
+    )
+      throw new Error(
+        'Only square or hexagonal planning unit grids are supported.',
+      );
+    if(
+      isNil(project.planningUnitAreakm2) ||
+      isNil(project.bbox)
+    ) throw new Error('Incomplete project data: this may be a bug.');
+
+    const [xmin, ymin, xmax, ymax] = project.bbox;
+    const query = `insert into scenarios_pu_data (pu_geom_id, scenario_id, puid)
```

Comment:
Shouldn't it remove previous ones?

Reply:
it shouldn't, no - once we create a scenario, the linking stays forever.

this can change if the user edits the parent project and changes planning area, grid shape or grid size, in which case we'll need to delete & gc all the scenarios, and when a user creates a new scenario in the edited project, we redo the linking here.